app/models/invitation.rb

```diff
@@ -60,12 +60,18 @@ def sent?
   def notify_invitee
     # TODO: maybe subclass to Invitation and ProviderInvitation
     if account.provider?
-      ProviderInvitationMailer.invitation(self).deliver_now
+      ProviderInvitationMailer.register_observer(InvitationDeliveryObserver)
```

Comment:
You should not use an observer, it is deprecated and we keep it for legacy. Work on removing it is a tech-debt task